### DISCUSSION TITLE
How to sort nodal values based on x, then y then z?

### DISCUSSION THREAD

**Discussion by aaelmeli** [idaholab/moose#25564] [Q&A General]
Hi,

I have a vector of sampled data on a plane from a 3D mesh where their sampled coordinates look like:

>     x     y     z
> 0.015, 0, 0.021
> 0.015, 0, 0.001
> 0.015, 0, 0.006
> 0.015, 0, 0.012
> 0.015, 0.001, 0.029
> 0.015, 0.001, 0.005
> 0.015, 0.001, 0.003
> 0.015, 0.001, 0.014

I used `NodalValueSampler` to sample the nodal values with `sort_by=y` option. As you see, `x` is fixed because I am sampling on a plane at `x=0.015`, `y` goes from lower to higher numbers (essentially because I used `sort_by=y`), however, for `z`, I could not sort it ascendingly like the following:

> 0.015, 0, 0.001
> 0.015, 0, 0.006
> 0.015, 0, 0.012
> 0.015, 0, 0.021
> 0.015, 0.001, 0.003
> 0.015, 0.001, 0.005
> 0.015, 0.001, 0.014
> 0.015, 0.001, 0.029

Now, is there a way to achieve this in moose?

**Comment by GiudGiud** [comment]
Hello

The logic in the sampler can only use a single ordering.
You ll have to hack a little bit. Create a ParsedFunction that does: y + 1e-6 * z, then use a FunctionIC to set an auxiliary variable sort_order, then sample it it as well and use it as the sort_by parameter

Guillaume

**Reply by GiudGiud** [reply]
I ll create a feature request for multiple orderings

**Reply by aaelmeli** [reply]
```
> [AuxVariables]
>   [sort_order]
>     order = FIRST
>     family = MONOMIAL
>   []
> []
> 
> [ICs]
>   [sortorder]
>     type = FunctionIC
>     variable = 'sort_order'
>     function = yz
>   []
> []
> 
> [Functions]
>   [yz]
>     type = ParsedFunction
>     value = 'y+1e-6*z'
>   []
> []
> 
> [VectorPostprocessors]
>   [wavefield_real_rcv]
>     type = NodalValueSampler
>     variable = 'uzr sort_order'
>     boundary = 'rcv_points'
>     sort_by = 'sort_order'
>     outputs = 'u_all'
>   []
> []
```
Is this what you meant? If yes, the problem is `sort_by` accept only `id, x, y, or z.`

**Reply by GiudGiud** [reply]
Ah that s unfortunate.

It should not be difficult to change though in the code. Look at samplerBase
